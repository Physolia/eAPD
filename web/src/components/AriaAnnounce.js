import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import { getAriaAnnouncement } from '../reducers/aria';

const AriaAnnounce = props => {
  const { ariaMessage } = props;
  return (
    <div
      className="ds-u-visibility--screen-reader"
      role="region"
      aria-live="polite"
    >
      {ariaMessage}
    </div>
  );
};

AriaAnnounce.propTypes = {
  ariaMessage: PropTypes.string
};

AriaAnnounce.defaultProps = {
  ariaMessage: ''
};

const mapStateToProps = state => ({
  ariaMessage: getAriaAnnouncement(state)
});

export default connect(mapStateToProps)(AriaAnnounce);

export { AriaAnnounce as plain };
