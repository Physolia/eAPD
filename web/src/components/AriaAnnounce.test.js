import { shallow } from 'enzyme';
import React from 'react';
import { plain as AriaAnnounce } from './AriaAnnounce';

describe('aria announce component', () => {
  it('renders correctly', () => {
    const component = shallow(<AriaAnnounce />);
    expect(component).toMatchSnapshot();
  });

  it('renders a message', () => {
    const props = {
      ariaMessage: 'No. No. No. No.'
    };
    const component = shallow(<AriaAnnounce props={props} />);
    expect(component).toMatchSnapshot();
  });
});
