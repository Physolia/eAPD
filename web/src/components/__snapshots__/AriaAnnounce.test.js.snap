// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aria announce component renders a message 1`] = `
<div
  aria-live="polite"
  className="ds-u-visibility--screen-reader"
  role="region"
/>
`;

exports[`aria announce component renders correctly 1`] = `
<div
  aria-live="polite"
  className="ds-u-visibility--screen-reader"
  role="region"
/>
`;
